src/Campaigns/Migrations/Donations/AddCampaignId.php

```diff
@@ -83,6 +88,14 @@ public function runBatch($firstId, $lastId)
 
             $donations = $query->getAll();
 
+            if (empty($donations)) {
+                return;
+            }
+
+            // Delete existing DonationMetaKeys::CAMPAIGN_ID records for these donations to prevent duplicates
+            $donationIds = implode(', ', array_column($donations, 'ID'));            
+            DB::query("DELETE FROM " . DB::prefix('give_donationmeta') . " WHERE donation_id IN ($donationIds) AND meta_key = '" . DonationMetaKeys::CAMPAIGN_ID . "'");
```

Comment:
@glaubersilva I don't think this migration should be concerned with deleting existing duplicates.  That would be a side effect of previously running this more than once.  The revers method will solve that.  

The purpose of the migration is to add campaign IDs however, we're using the DB insert method on the meta table which if re-run would add the duplicate.  

So would suggest simply updating the insert meta sql to either use the built-in WP update_meta function or do a `ON DUPLICATE KEY UPDATE`

<img width="525" alt="Screenshot 2025-07-09 at 11 44 25 AM" src="https://github.com/user-attachments/assets/ccc5d1d3-c54c-42ff-bc66-4e4d9bdaece1" />

Reply:
@jonwaldstein  Oh cool! I didn't know about the `REPLACE INTO`. 

About your questions:

1. Yes, the bulk delete is made for the donations from the current batch only.
2. Yes, it is necessary because the reverse button will only be displayed in cases where the migration failed, while this delete logic will prevent the duplicate entries when someone re-runs it.

So, I think it is better to keep both functionalities because for the support team, it would be easier to solve the problem in sites that already have the duplicated entries, as all they need to do is re-run this migration, and everything should be fixed. In other words, they won't need to manually change the status of the migration on the DB to "failed" to see the reverse button.

Reply:
@alaca I agree with you about the purpose of the runBatch function.  It's purpose is to insert the campaign ID.  I'm still tempted to treat this one as a special use case because of how important it is.  On the rollback topic - I thought we chose not to allow rollback of successful migration because it could cause weird side effects if being reversed out of order with other migrations.

Reply:
No, we don't need to do that if we enable the rollback option for successful migration.

Reply:
@jonwaldstein @alaca Resolved in dbad2fc9d6b4147483005c1aa60196b4f46bf2f0

As discussed in the call today, I removed the logic to delete meta keys from the `runBatch` method and enabled the `Reverse Update` button even when the migration has a success status and the final result is this one:

<img width="1715" height="69" alt="image" src="https://github.com/user-attachments/assets/f0bbcac3-abaf-4efe-9775-0180c2f613cb" />

Both buttons, `Re-Run Update` and `Reverse Update`, are displayed for successful migrations that implemented the ReversibleMigration interface.

This is the expected behavior, right?